Feature/settings 메인 색상 및 CSS 전면 수정, 메인페이지 배너 추가, 커뮤니티 레이아웃 구현, QR코드 생성

sen2y opened the pull request:
## 📝 개요

- 1차 배포를 위해 막아놓았던 페이지들의 주석 제거
- 서비스 메인색상 및 주요 디자인 변경, + 메인페이지 배너 추가
- 커뮤니티 레이아웃 추가 ( 게시글 작성,조회,수정,삭제,검색 ) 
- sentry 에러 리포트 등록 버튼 추가

## ✨ 변경 사항

- 코드나 기능의 주요 변경 사항을 설명합니다.

  - ✨ 메인 색상 변경, 메인페이지에 여행 배너 추가( 클릭안됨, 광고용배너 )
  - ✨ 커뮤니티 관련 레이아웃 구현 완료 - API 기능개발 완료시 연결 예정
  - ✨ 동행 채팅방 내 햄버거바 Drawer 레이아웃 - Tab으로 분리, 디테일 구현사항은 다음 PR에 추가될 예정입니다.

## 🔗 관련 이슈

- 이 PR과 관련된 이슈 번호를 연결합니다. (없으면 생략)
  - 예: #51, #54, #50, #40,  #48, #58, #59
  - 
## 📸 스크린샷 (옵션)
<img width="300" alt="image" src="https://github.com/user-attachments/assets/f8cffc8c-7f8e-4687-b275-6f39b4a5af66">
<img width="300" alt="image" src="https://github.com/user-attachments/assets/288e267e-7ce9-44e6-9c16-20f7fc8fd3c3">
<img width="300" alt="image" src="https://github.com/user-attachments/assets/ee67c615-7a48-4174-b738-b4e8f46c08a4">
<img width="300" alt="image" src="https://github.com/user-attachments/assets/3edada22-4b73-49dd-a7f7-bc8272beee8f">
<img width="300" alt="image" src="https://github.com/user-attachments/assets/0a0c3a89-4463-4412-b9ff-323b9f8af4a5">
<img width="300" alt="image" src="https://github.com/user-attachments/assets/6c1ec8f9-bdb7-42a5-90ec-6d09f6c92c83">


- UI 변경 사항이 있는 경우 스크린샷을 포함합니다.

## ℹ️ 참고 사항

- 리뷰어가 알 필요가 있는 추가 정보나 문서, 참고 링크를 포함합니다.

sen2y committed:
Merge pull request #44 from 100-hours-a-week/dev

🚀 Release: v1.0.0

sen2y committed:
💄 메인 컬러 변경

sen2y committed:
🎨 배너 svg 추가

sen2y committed:
🐛 모바일 화면 인풋 클릭시 확대되는거 방지

sen2y committed:
✨ 광고 배너 추가

sen2y committed:
💄 레이아웃 css 변경

sen2y committed:
🎨 배너 이미지 변경

sen2y committed:
✨ 채팅방 상세 페이지 주석 복구, 상단에 햄버거바 추가

sen2y committed:
✨ QR코드 생성 추가

sen2y committed:
♻️ 주석 코드 복구 및 기타 코드 업데이트

sen2y committed:
✨ 채팅방 페이지 내 햄버거바 Drawer, 내부 Tab 구현

sen2y committed:
💄 반응형 w 기준 768 -> 500px

sen2y committed:
💄 css update - 색상 변경, carousel 드래그 허용

sen2y committed:
✨ 커뮤니티 목록 페이지 구현

sen2y committed:
✨ 커뮤니티 상세 페이지 구현

sen2y committed:
💄 화면 테두리 그림자 추가

sen2y committed:
♻️ 동행, 커뮤니티 목록조회 페이지 컴포넌트 리팩토링

sen2y committed:
🔧 센트리 에러 리포트 버튼 추가

sen2y committed:
💄 css 틀 깨지는 이슈 해결

sen2y committed:
💄  css update

sen2y committed:
🐛 ci build 에러 요소 제거

Namgyu11 reviewed (approved):
고생 많으셨습니다 ! 
LGTM 😎😎

Namgyu11 commented on src/app/components/community/CommunityCard.tsx:
PostCard와 CommunityCard 컴포넌트는 거의 동일한 구조를 가지고 있는 것 같습니다!
공통된 코드를 추상화해서  재사용 가능한 하나의 컴포넌트로 만드는 것은 어떨까용?

Namgyu11 reviewed (approved):
UI/UX가 더 깔끔하고 보기 좋아진 것 같아요!!
고생 많으셨습니다!!
LGTM😎😎

TopazKang reviewed (approved):
코드 확인 완료하였습니다. 고생 많으셨습니다!

sen2y committed:
➕ ✨ QR코드 이미지로 변환 및 복사 로직 추가 ( html2canvas )

sen2y committed:
🐛 fix: 모바일 댓글 깨짐 이슈 해결

- 반응형으로 댓글 입력창과 등록 버튼의 너비를 퍼센트 기반으로 조정하여 모바일 화면에서 깨짐 현상 해결

closes 48

sen2y committed:
🐛  모바일 뷰포트 설정 추가
- 모바일 뷰포트 설정을 통해 화면 확대 방지
- Head 컴포넌트를 사용하여 메타 태그 설정

sen2y committed:
♻️ 채팅방에는 헤더 노출 제외 + 뒤로가기 버튼 추가 ( router.back() )

sen2y committed:
♻️ css 및 코드 Update

sen2y committed:
🔥 안쓰는 코드 삭제

sen2y committed:
💄 css update - 전송버튼과 focus 색상 통일

sen2y committed:
🔥 콘솔 에러메시지 요인 해결

sen2y committed:
⚡️ feat: Safari에서 클립보드 복사 기능 호환성 개선

- Safari에서 클립보드 API 사용 시 발생하는 오류를 해결하기 위해 비동기 처리 방식 개선
- QR 코드 복사 기능을 Safari에서도 정상적으로 동작하도록 수정

kimsungho97 commented on src/app/components/accompany/PostList.tsx:
```diff
@@ -0,0 +1,31 @@
+  PostComponent,
+}: PostListProps<T>) => {
+  return (
+    <div className='container mx-auto mt-4 mb-10'>
+      {posts.length > 0 ? (
+        posts.map((post) => (
+          <div key={post.id} onClick={() => onPostClick(post.id)}>
+            <PostComponent {...post} />
+          </div>
+        ))
```
이 파일이 재활용될 수 있을 지는 모르겠지만..
제 개인적인 취향으로는 이런 형태의 구조 좋아합니다ㅋㅋㅋㅋ굳굳

kimsungho97 commented on src/app/components/chat/MenuDrawer.tsx:
```diff
@@ -0,0 +1,60 @@
+const MenuDrawer: React.FC = () => {
+  const [open, setOpen] = useState(false)
+
+  const showDrawer = () => {
+    setOpen(true)
+  }
+
+  const onClose = () => {
+    setOpen(false)
+  }
+
+  return (
+    <>
+      <div
+        className='pl-5 text-secondary hover:text-black cursor-pointer'
+        onClick={showDrawer}
+      >
```
이건 UX 또는 기획 관점으로 넘어갈 만한 주제인데요,
Drawer버튼은 toggle방식으로 동작해도 될것같습니다. 위의 showDrawer, onClose 를 사용하지 않고 `setOpen((o)=>!o);`같은 방법으로 구현해도 좋을것같습니다

kimsungho97 commented on src/app/components/common/ShareModal.tsx:
```diff
@@ -17,9 +19,31 @@ const ShareModal: React.FC<ShareModalProps> = ({
   const defaultUrl = `${process.env.NEXT_PUBLIC_SELF_URL}${pathname}`
+  const qrCodeRef = useRef<HTMLDivElement>(null)
+
+  const handleCopyQRCode = async () => {
+    if (qrCodeRef.current) {
+      try {
+        const canvas = await html2canvas(qrCodeRef.current)
+        canvas.toBlob(async (blob) => {
+          if (blob) {
+            try {
+              const item = new ClipboardItem({ 'image/png': blob })
+              await navigator.clipboard.write([item])
+              showSuccess('QR 코드가 클립보드에 복사되었습니다!')
+            } catch (error) {
+              showSuccess('클립보드에 복사할 수 없습니다: ' + error)
+            }
+          }
+        })
+      } catch (error) {
+        showSuccess(`${error}, QR 코드를 복사하는 중 오류가 발생했습니다.`)
+      }
+    }
+  }
```
저번에 클립보드 관련해서 safari 오류 개선하신거 본거같은데, 클립보드 복사기능은 브라우저별로 테스트해보시는게 좋을것같습니다
